Source-Code/SettingsGui.cs

```diff
@@ -173,18 +181,26 @@ public void Awake()
             }
 
             // Load and configure once
-            if (HighLogic.LoadedScene == GameScenes.SPACECENTER || HighLogic.LoadedScene == GameScenes.FLIGHT)
-            {
+            if (HighLogic.LoadedScene == GameScenes.SPACECENTER || HighLogic.LoadedScene == GameScenes.FLIGHT)
+            {
+
+	            if (onlyInSpaceCenter && HighLogic.LoadedSceneIsFlight)
+	            {
+		            ApplicationLauncher.Instance.RemoveModApplication(appLauncherButton);

```

Comment:
Have you tested repeatedly going from flight to space center with this code?  There seems to be some strange behavior with removing launcher buttons in KSP 1.0.  I think it might be better to hide the button during GameScenes.FLIGHT by changing appLauncherButton.VisibleInScenes when onlyInSpaceCenter changes, although I haven't tested that code.